gyp/pylib/gyp/input.py

```diff
@@ -979,7 +979,7 @@ def ExpandVariables(input, phase, variables, build_file):
                     p_stdout = p_stdout.decode("utf-8")
                     p_stderr = p_stderr.decode("utf-8")
 
-                    if p.wait() != 0 or p_stderr:
+                    if p.wait() != 0:
                         sys.stderr.write(p_stderr)
                         # Simulate check_call behavior, since check_call only exists
                         # in python 2.5 and later.
```

Comment:
```suggestion
                    if p_stderr:
                        sys.stderr.write(p_stderr)
                    if p.wait() != 0:
```

Reply:
Is there a simple way that this code can detect the presence of the debugger?

Reply:
If we are going to make changes here, it is probably best to upgrade to [`subprocess.run()`](https://docs.python.org/3/library/subprocess.html) to remove a lot of clutter.

Reply:
keep comments ? (keep/delete) ?
```
                        # Simulate check_call behavior, since check_call only exists
                        # in python 2.5 and later.
```

this is better behavior, no loss of stderr output (possible useful debug log)

Reply:
We do not need the comment about Py25...  https://docs.python.org/3/library/subprocess.html#subprocess.check_call is fully available to us on all supported versions of Python and the even simpler `subprocess.run()` is too.

Will _all_ users have the debugger present?  If not, what is the effect of the proposed changes on those users?

Reply:
> what is the effect of the proposed changes on those users?

`<!(node -p \"require('node-addon-api').gyp\")`

oof, the scope is Any command..

___

> Is there a simple way that this code can detect the presence of the debugger?

there may be another way: silence the:
```
Debugger attached.
Waiting for the debugger to disconnect...
```

https://github.com/nodejs/node/blob/fddc701d3c0eb4520f2af570876cc987ae6b4ba2/src/inspector_agent.cc#L802-L805
It seems that it only prints if no parent
python spawning seems like a parent ? I'll have to check more

Reply:
Stderr doesn’t mean error,
Status code > 0 means error

> Yes, programs can output to standard error and still return a zero exit status. You will find many POSIX utilities specifications which state that "The standard error shall be used only for diagnostic messages."

https://unix.stackexchange.com/a/594670

this is the correct handling

If any program wants to indicate error, just set status code